### ISSUE TITLE
Schema derived data is computed twice

### ISSUE THREAD

**Issue by tovbinm** [apollographql/apollo-server#5729]
Hello,

When profiling our service application (below) I noticed that Apollo Server generates schema derived data twice. Since it's a time consuming operation I would assume it should only be happening once. For the context here is our server code snippet and profiler chart:
```typescript
export async function createViewMiddleware(params: ViewMiddlewareParams): Promise<ApolloMiddleware> {
	  const server = new ApolloServer({
	      schema: params.graphqlSchema,
	      plugins: params.plugins,
	      introspection: true,
	      mocks: false,
	      mockEntireSchema: false,
	      persistedQueries: false,
	      stopOnTerminationSignals: true,
	  });
	  await server.start();
	  return { middleware: toKoaMiddleware(server), server };
}
```
<img width="318" alt="Screen Shot 2021-09-20 at 10 23 56 PM" src="https://user-images.githubusercontent.com/629845/134116022-3afbb143-b2bd-4cd7-9d0b-2d1e3e819f94.png">

**Comment by glasser** [comment]
@sachindshinde Can you take a quick look at this? I think this might be expected due to the changes we made with SchemaManager and maybe it's avoidable?  (Note that they are using `{schema}`, not gateway.

**Comment by sachindshinde** [comment]
@glasser It turned out to be avoidable without too much issue, PR to fix at #5757 .
